File: pet-app/lib/features/pet/domain/pet_entity.dart
class PetEntity {
  final String id;
  final String userId;
  final String name;
  final String species;
  final String? breed;
  final DateTime? dateOfBirth;
  final double? weightKg;
  final String? medicalHistory;
  final String? avatarUrl;
  final DateTime createdAt;

  const PetEntity({
    required this.id,
    required this.userId,
    required this.name,
    required this.species,
    this.breed,
    this.dateOfBirth,
    this.weightKg,
    this.medicalHistory,
    this.avatarUrl,
    required this.createdAt,
  });

  int get age {
    if (dateOfBirth == null) return 0;
    return DateTime.now().year - dateOfBirth!.year;
  }

  String get ageGroup {
    if (species == 'cat') {
      if (age > 10) return 'Senior';
      if (age < 1) return 'Kitten';
      return 'Adult';
    }
    if (age > 8) return 'Senior';
    if (age < 1) return 'Puppy';
    return 'Adult';
  }
}

File: pet-backend/src/modules/pets/pet.entity.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  ManyToOne,
  JoinColumn,
  CreateDateColumn,
  UpdateDateColumn,
} from 'typeorm';
import { User } from '../users/user.entity';

@Entity('pets')
export class Pet {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @ManyToOne(() => User)
  @JoinColumn({ name: 'userId' })
  user: User;

  @Column()
  userId: string;

  @Column()
  name: string;

  @Column()
  species: string;

  @Column({ nullable: true })
  breed: string;

  @Column({ type: 'date', nullable: true })
  dateOfBirth: Date;

  @Column({ type: 'decimal', precision: 5, scale: 2, nullable: true })
  weightKg: number;

  @Column({ type: 'text', nullable: true })
  medicalHistory: string;

  @Column({ nullable: true })
  avatarUrl: string;

  @CreateDateColumn()
  createdAt: Date;

  @UpdateDateColumn()
  updatedAt: Date;
}
